### DISCUSSION TITLE
Limit number of parallel tests

### DISCUSSION THREAD

**Discussion by llRandom** [xunit/xunit#2670] [Question] [answered]
How to limit number of parallel tests running? I know there's a `maxParallelThreads` option but it's not the same. With 2 parallel threads xunit may start hundreds of parallel tests. I know I can also turn off `parallelizeTestCollections` but it's "all or nothing" option. The reason I need to limit it:
1. I have some test(s) that keeps hanging randomly. I've tried to use `longRunningTestSeconds` option to figure one the cause but with hundreds of parallel tests all of them are logged. And when I disable parallel test execution it's not reproducible but it's slow.
2. Some tests have datetime asserts with precision threshold and with number of tests growing I have to keep increasing the threshold
3. Lots of tests are using Asp.net core test server with http client. I have to keep increasing http client timeout with number of parallel tests growing and I want to avoid increasing it further to catch hanging tests
There might be some cases when it makes sense to run more tests that number of parallel threads but usually there's no point to run hundreds of tests at the same time

**Comment by bradwilson** [comment, marked as answer]
There is no mechanism today to limit the number of _started_ tests, only to limit the number of _concurrently running_ tests. There are currently no plans for this functionality.
